Zarr tests failing with Xarray >= 2025.06.0

Right now unrelated dependency updates such as #314 are hitting errors when running Zarr tests. https://github.com/xpublish-community/xpublish/actions/runs/20640684125/job/59271910173?pr=314#step:6:5002


```py
     def create_zmetadata(dataset: xr.Dataset) -> dict:
...
>           encoding = extract_zarr_variable_encoding(
                dvar,
                zarr_format=ZARR_FORMAT,
                region=tuple([SimpleNamespace(start=None, stop=None) for _ in da.shape]),
            )
E           TypeError: extract_zarr_variable_encoding() got an unexpected keyword argument 'region'

xpublish/utils/zarr.py:202: TypeError
```

https://github.com/xpublish-community/xpublish/blob/bec4efa576507d7fb15ce1a4fc7b8ddec6298ad8/xpublish/utils/zarr.py#L181-L216

It looks like the signature of the function changed in https://github.com/pydata/xarray/pull/10336

--- a/requirements.txt
+++ b/requirements.txt
@@ -7,5 +7,5 @@ pluggy
 pydantic>=2.5.0
 toolz
 uvicorn
-xarray>=2025.1.0
+xarray>=2025.7.0
 zarr>=3.0.7

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -51,7 +51,7 @@ def test_invalid_encoding_chunks_with_dask_raise():
     with pytest.raises(ValueError) as excinfo:
         _ = create_zmetadata(ds)
     excinfo.match(
-        r"Specified zarr chunks encoding\['chunks'\]=\(8, 5, 1\) for variable named None would overlap multiple dask chunks*"
+        r"Specified Zarr chunks encoding\['chunks'\]=\[8, 5, 1\] for variable named 'foo' would overlap multiple Dask chunks.*"
     )
 
 

--- a/xpublish/utils/zarr.py
+++ b/xpublish/utils/zarr.py
@@ -17,6 +17,7 @@ import numpy as np
 import xarray as xr
 from numcodecs.abc import Codec
 from numcodecs.compat import ensure_ndarray
+from xarray.backends.chunks import validate_grid_chunks_alignment
 from xarray.backends.zarr import (
     DIMENSION_KEY,
     encode_zarr_attr_value,
@@ -199,10 +200,18 @@ def create_zmetadata(dataset: xr.Dataset) -> dict:
             dvar = da.variable
 
         encoded_da = encode_zarr_variable(dvar, name=key)
+        if 'chunks' in encoded_da.encoding:
+            validate_grid_chunks_alignment(
+                nd_v_chunks=dvar.chunks,
+                enc_chunks=encoded_da.encoding['chunks'],
+                region=tuple([SimpleNamespace(start=None, stop=None) for _ in da.shape]),
+                allow_partial_chunks=False,
+                name=key,
+                backend_shape=encoded_da.shape,
+            )
         encoding = extract_zarr_variable_encoding(
             dvar,
             zarr_format=ZARR_FORMAT,
-            region=tuple([SimpleNamespace(start=None, stop=None) for _ in da.shape]),
         )
         zattrs = _extract_dataarray_zattrs(encoded_da)
         zattrs = _extract_dataarray_coords(da, zattrs)

